src/components/data-management/MetadataPopover.jsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Popover } from 'antd';
+import EditableField from '../EditableField';
+import { metadataNameToKey } from '../../utils/metadataUtils';
+
+const MetadataPopover = (props) => {
+  const {
+    existingMetadata,
+    popoverPosition,
+    onCreate,
+    onCancel,
+    message,
+    children,
+    ...restOfProps
+  } = props;
+
+  const getContent = () => (
+    <EditableField
+      onAfterSubmit={(value) => {
+        onCreate(value);
+      }}
+      onAfterCancel={() => {
+        onCancel();
+      }}
+      deleteEnabled={false}
+      value='Track name'
+      defaultEditing
+      validationFunc={(value) => value.length > 0 && !existingMetadata.includes(metadataNameToKey(value))}
+    />
+  );
+
+  const content = getContent();
+
+  let style = {};
+  if (popoverPosition) {
+    style = { position: 'absolute', left: popoverPosition.current.x + 20, top: popoverPosition.current.y + 20 };
+  }
+
+  /* eslint-disable react/jsx-props-no-spreading */
```

Comment:
I don't think we need this `eslinit` comment, do we? 

Additionally, is the whole `if` required? Does {children} break if it is empty? And, if it does, could we go with a `? :` ?

Reply:
The eslint is required to supress props spreading in the `if` and `return` block. 

Yea, the `if` block isn't required, I'll remove it